Userland/Applications/Presenter/SlideObject.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2022, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Forward.h>
+#include <AK/NonnullOwnPtr.h>
+#include <AK/NonnullRefPtr.h>
+#include <LibCore/Object.h>
+#include <LibGUI/MessageBox.h>
+#include <LibGUI/Window.h>
+#include <LibGfx/Bitmap.h>
+#include <LibGfx/Color.h>
+#include <LibGfx/Font/Font.h>
+#include <LibGfx/Forward.h>
+#include <LibGfx/Painter.h>
+#include <LibGfx/Rect.h>
+#include <LibGfx/TextAlignment.h>
+#include <LibImageDecoderClient/Client.h>
+
+// Anything that can be on a slide.
+// For properties set in the file, we re-use the Core::Object property facility.
+class SlideObject : public Core::Object {
```

Comment:
Perhaps I'm missing something, but why are these SlideObjects not just GUI::Widgets? Then you could override the paint_event instead of inventing your own virtual method.

Reply:
The heirarchy of "not widgets, slide objects" feels very close to existing GUI::Widgets

Reply:
SlideObjects don't participate in layouting or LibGUI communication, they just use LibGfx painting APIs. They are only objects so that they can use properties, which greatly simplifies and modularizes the object property setting. Also, not all parts of the slide system are Core::Objects, e.g. not the slide or the presentation themselves.